api_endpoints.yaml

```diff
@@ -0,0 +1,1732 @@
+        required: true
+        type: "string"
+        items:
+          type: "string"
+      responses:
+        "200":
+          description: "successful operation"
+          schema:
+            type: "array"
+            items:
+              $ref: "#/definitions/Profile"
+            example:
+              testProfile1:
+                value:
+                  profile_id: 10
+                  email: "test@email.com"
+                  name: "bob"
+                  auth0_id: "234234923742983"
+                  role: "admin"
+                  avatarUrl: "profileImageUrl"
+                  pending: "accountStatus"
+              testProfile2:
+                value:
+                  profile_id: 10
+                  email: "test@email.com"
+                  name: "bob"
+                  auth0_id: "234234923742983"
+                  role: "admin"
+                  avatarUrl: "profileImageUrl"
+                  pending: "accountStatus"     
+        "400":
+          description: "Invalid role_id"
+        "404":
+          description: "no profiles associated with that role_id."  
+      
+  /user:
+    get:
+      security:
+        - userAuth: []
+      tags:
+      - "user"
+      summary: "Returns an event object with the specified type"
+      description: "This can only be done by the logged in user."
+      operationId: "findUserData"
+      produces:
+      - "application/json"
+      responses:
+        default:
+          description: "successful operation"
+  /user/inbox:
+    get:
+      security:
+        - userAuth: []
+      tags:
+      - "user"
+      summary: "Returns an event object"
+      description: "This can only be done by the logged in user."
+      operationId: "getInbox"
+      produces:
+      - "application/json"
+      responses:
+        default:
+          description: "successful operation"
+  /user/schedules:
+    get:
+      security:
+        - userAuth: []
+      tags:
+      - "user"
+      summary: "Returns an schedule object"
+      description: "A list of enrolled courses_id's with start and end times."
+      operationId: "getSchedule"
+      produces:
+      - "application/json"
+      responses:
+        "200":
+          description: "successful operation"
+  /user/{user_id}:
+    put:
+      security:
+        - userAuth: []
+      tags:
+      - "user"
+      summary: "Updates specific user"
+      description: "Returns an event object with the specified user_id. Updates specific user"
+      operationId: "updateUserData"
+      produces:
+      - "application/json"
+      parameters:
+      - name: "user_id"
+        in: "path"
+        description: "The logged-in user_id"
+        required: true
+        type: "string"
+      responses:
+        "200":
+          description: "successful operation"
+          schema:
+            $ref: "#/definitions/User"
+        "400":
+          description: "invalid user id"
+        "404":
+          description: "user not found"  
```

Comment:
Not sure if we need this endpoint anymore since we are using Auth0. We might want to consider shifting any previous user endpoints into the profile endpoints. More clarification is needed on whether having user endpoints is a net benefit.